tests/legacy-cli/e2e/tests/build/auto-csp.ts

```diff
@@ -0,0 +1,131 @@
+import assert from 'node:assert';
+import { getGlobalVariable } from '../../utils/env';
+import { expectFileToMatch, rimraf, writeMultipleFiles } from '../../utils/fs';
+import { findFreePort } from '../../utils/network';
+import { installWorkspacePackages } from '../../utils/packages';
+import { execAndWaitForOutputToMatch, ng } from '../../utils/process';
+import { updateJsonFile, updateServerFileForWebpack, useSha } from '../../utils/project';
+
+const MULTI_HASH_CSP =
+  /script-src 'strict-dynamic' (?:'sha256-[^']+' )+https: 'unsafe-inline';object-src 'none';base-uri 'self';/;
```

Comment:
At this level of abstraction I wouldn't assert on the policy this precisely, that's more of a unit test level assertion. Instead I'd just check:
1. That a CSP policy was added (regex match for `<meta http-equiv="Content-Security-Policy"`), don't care what the policy is.
2. That the browser did not error from that policy.

Optionally it might also be good to verify that the scripts actually executed (maybe just check that they printed to the console?). That way we know we aren't asserting too soon before the scripts have been checked, I think there's a low risk of a false positive there though, so probably not too important.

Reply:
I think verifying that there are no errors should cover the script execution cases but to ensure i think it would be good to check the console output for the added script logging (`Inline Script...`) cases as well.

Reply:
Updated to just check for `const CSP_META_TAG = /<meta http-equiv="Content-Security-Policy"/;` and to check that all four scripts ran in order (two inline and two sourced) with globally set variables from previously run scripts.